crypto/s2n_hash.c

```diff
@@ -557,7 +557,7 @@ int s2n_hash_update(struct s2n_hash_state *state, const void *data, uint32_t siz
 int s2n_hash_digest(struct s2n_hash_state *state, void *out, uint32_t size)
 {
     PRECONDITION_POSIX(s2n_hash_state_validate(state));
-    ENSURE_POSIX(S2N_MEM_IS_READABLE(out, size), S2N_ERR_PRECONDITION_VIOLATION);
+    ENSURE_POSIX(size == 0 || S2N_MEM_IS_READABLE(out, size), S2N_ERR_PRECONDITION_VIOLATION);
```

Comment:
Why is this new condition added? The macro is defined as
```
#define S2N_MEM_IS_READABLE(base, len) (((len) == 0) || (base) != NULL)
```
which already includes this new one, right?

Reply:
True! I removed it.